Assets/Scripts/Player/PlayerController.cs

```diff
@@ -160,6 +218,17 @@ private void OnRightClick(InputAction.CallbackContext obj)
                     _attackLogic.StopAttack();
                     Move(groundHit);
                 }
+        
```

Comment:
that's kinda unclear. If there is no damageable, then move, after that you check for an interactable (when it's already moving). It should be if there is no damageable, check for an interactable, if there is no interactable, then move. You should include moving towards the interactable in your InteractLogic class like done for attacking.
I suggest something like:
```
Interactable interactable = objectHit.GetComponent<Interactable>();
IDamageable damageable = objectHit.GetComponent<IDamageable>();

if (damageable != null)
{
     _interactLogic.RemoveFocus();
     _attackLogic.StartAttack(objectHit);
}
else if (interactable != null)
{
     _attackLogic.StopAttack();
     _interactLogic.SetFocus(interactable);
}
else if (Physics.Raycast(clickRay, out RaycastHit groundHit, 10000, moveClickLayers))
{
     _attackLogic.StopAttack();
     _interactLogic.RemoveFocus();
     Move(groundHit);
}

```

Reply:
this looks good to me.

Reply:
Thanks for implementing! with interactlogic it looks much better 👍